packages/cdk/lib/stacks/engines/toil-engine-construct.ts

```diff
@@ -0,0 +1,104 @@
+import { Aws } from "aws-cdk-lib";
+import { IVpc } from "aws-cdk-lib/aws-ec2";
+import { FargateTaskDefinition, LogDriver } from "aws-cdk-lib/aws-ecs";
+import { ApiProxy, SecureService } from "../../constructs";
+import { IRole } from "aws-cdk-lib/aws-iam";
+import { createEcrImage, renderServiceWithTaskDefinition } from "../../util";
+import { Bucket } from "aws-cdk-lib/aws-s3";
+import { EngineOptions, ServiceContainer } from "../../types";
+import { LogGroup, ILogGroup } from "aws-cdk-lib/aws-logs";
+import { EngineOutputs, EngineConstruct } from "./engine-construct";
+import { ToilJobRole } from "../../roles/toil-job-role";
+import { ToilEngineRole } from "../../roles/toil-engine-role";
+import { IJobQueue } from "@aws-cdk/aws-batch-alpha";
+import { Construct } from "constructs";
+
+export interface ToilEngineConstructProps extends EngineOptions {
+  /**
+   * AWS Batch JobQueue to use for running workflows.
+   */
+  readonly jobQueue: IJobQueue;
+}
+
+export class ToilEngineConstruct extends EngineConstruct {
+  public readonly engine: SecureService;
+  public readonly adapterRole: IRole;
+  public readonly apiProxy: ApiProxy;
+  public readonly adapterLogGroup: ILogGroup;
+  public readonly engineLogGroup: ILogGroup;
+  public readonly engineRole: IRole;
+  public readonly jobRole: IRole;
+
+  constructor(scope: Construct, id: string, props: ToilEngineConstructProps) {
+    super(scope, id);
+    const params = props.contextParameters;
+    this.engineLogGroup = new LogGroup(this, "EngineLogGroup");
+    const artifactBucket = Bucket.fromBucketName(this, "ArtifactBucket", params.artifactBucketName);
+    const outputBucket = Bucket.fromBucketName(this, "OutputBucket", params.outputBucketName);
+
+    // Make a role for the jobs to run as
+    this.jobRole = new ToilJobRole(this, "ToilJobRole", {
+      readOnlyBucketArns: (params.readBucketArns ?? []).concat(artifactBucket.bucketArn),
+      readWriteBucketArns: (params.readWriteBucketArns ?? []).concat(outputBucket.bucketArn),
+      policies: props.policyOptions,
+    });
+
+    // Make a role for the server to run as
+    this.engineRole = new ToilEngineRole(this, "ToilEngineRole", {
+      jobQueueArn: props.jobQueue.jobQueueArn,
+      jobRoleArn: this.jobRole.roleArn,
+      readOnlyBucketArns: (params.readBucketArns ?? []).concat(artifactBucket.bucketArn),
+      readWriteBucketArns: (params.readWriteBucketArns ?? []).concat(outputBucket.bucketArn),
+      policies: props.policyOptions,
+    });
+
+    // Make the container and pass it the ARN of the role to use for individual jobs.
+    const engineContainer = params.getEngineContainer(props.jobQueue.jobQueueArn, {
+      TOIL_AWS_BATCH_JOB_ROLE_ARN: this.jobRole.roleArn,
+    });
+
+    // TODO: Move log group creation into service construct and make it a property
```

Comment:
Is there anything blocking this from happening right now? If not either move it or remove the TODO

Reply:
This comment came along with the other code I [duplicated from the Cromwell engine](https://github.com/aws/amazon-genomics-cli/blob/75603a860ca401b7be2d5d04c51248a227283d00/packages/cdk/lib/stacks/engines/cromwell-engine-construct.ts#L52-L54).

I could probably jut drop it here. There won't be an adapter log group to move if I manage to defeat the type system and get rid of it.